src/slang/parser.ts

```diff
@@ -163,43 +146,72 @@ const createAcornParserOptions = (context: Context): AcornOptions => ({
   }
 })
 
-rules.forEach(rule => {
-  const keys = Object.keys(rule.checkers)
-  keys.forEach(key => {
-    walkers[key] = compose(walkers[key], (node, context) => {
-      if (typeof rule.disableOn !== 'undefined' && context.chapter >= rule.disableOn) {
-        return
+function createWalkers(
+  allowedSyntaxes: { [nodeName: string]: number },
+  parserRules: Array<Rule<es.Node>>
+) {
+  const newWalkers: { [name: string]: (node: es.Node, ctxt: Context) => void } = {}
```

Comment:
Is it possible to use ES6 Map here? since you do use the `[]` member notation to assign functions.

Reply:
`Array.from(x.entries()).reduce((obj, pair) => {obj[pair[0]] = pair[1]; return obj;}, {})`
should be able to convert from Map to Object